currency-rates/internal/app/app.go

```diff
@@ -35,24 +51,17 @@ func (app *App) Run(ctx context.Context) error {
 		return err
 	}
 
-	subscriptionSender := services.NewEmail(app.cfg.EmailAddress, app.cfg.EmailPass, app.l)
-	subscriptionService := services.NewSubscriptionService(app.l, adminRepository, subscriptionSender, rateService)
+	emailSender, err := services.NewEmail(app.cfg.EmailAddress, app.cfg.EmailPass, app.l)
+	if err != nil {
+		return err
+	}
+
+	subscriptionService := services.NewSubscriptionService(app.l, adminRepository, emailSender, rateService)
 	ratesHandler := handlers.NewRateHandler(app.l, rateService)
 	subscriptionHandler := handlers.NewSubscribeHandler(app.l, adminRepository)
 
-	s, _ := gocron.NewScheduler()
+	s := app.initCron(ctx, subscriptionService)
 	defer func() { _ = s.Shutdown() }()
-	
-	_, _ = s.NewJob(
-		gocron.CronJob(
-			"0 2 * * *",
-			false,
-		),
-		gocron.NewTask(
-			subscriptionService.NotifySubscribers, ctx,
-		),
-	)
-
 	s.Start()
 
 	r := chi.NewRouter()
```

Comment:
Try to make separate constructor for this. This is not the responsibility of Run function, cause here you are just registering handlers, not running. Also try to imagine you are having hundred of route (that is normal for small-medium project) you Run will be a nightmare  for everyone reading your code =)

Reply:
Done